### PR TITLE
fix(pnpm): replace npm with pnpm in dockerfile

### PR DESCRIPTION
**Pull request by sammcj**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimisation
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [ ] Yes
- [x] No, because: it's for a version currently in RC

      
## Have you updated all relevant documentation?
- [x] Yes
- [ ] No

## Description

3.5.0 rc1 Docker builds are broken due to the swap from npm to pnpm.

I think this should fix it.

## Related Tickets & Documents

- Related Issue #5253 #5258
- Closes #

## QA Instructions, Screenshots, Recordings

Run docker build and tests as normal.

## Merge Plan

- Can be merged when approved and docker builds pass.

## Added/updated tests?

- [ ] Yes
- [x] No : not relevant?

### CODE REVIEW

**Comment by antage** [comment]
More pnpm-way version:

```
FROM node:18-slim AS web-builder
ENV PNPM_HOME="/pnpm"
ENV PATH="$PNPM_HOME:$PATH"
RUN corepack enable

WORKDIR /build
COPY invokeai/frontend/web/ ./
RUN --mount=type=cache,target=/pnpm/store \
    pnpm install --frozen-lockfile
RUN pnpm run build
```

**Comment by sammcj** [comment]
Ah I see, OK thanks updated.

File: docker/Dockerfile [hunk 1]
```diff
@@ -59,15 +59,19 @@ RUN --mount=type=cache,target=/root/.cache/pip \
...
+
 WORKDIR /build
 COPY invokeai/frontend/web/ ./
-RUN --mount=type=cache,target=/usr/lib/node_modules \
-    npm install --include dev
+RUN --mount=type=cache,target=/pnpm/store \
+    pnpm install --frozen-lockfile
+RUN pnpm run build
 RUN --mount=type=cache,target=/usr/lib/node_modules \
     yarn vite build
```
Comment (antage) [inline docker/Dockerfile:73]: It doesn't need `yarn vite build`, `pnpm run build` replaced it.

**Review by antage — APPROVED** [review]

**Comment by sammcj** [comment]
FYI 3.5.0 builds are still failing on rc3, this _should_ fix it.

**Review by hipsterusername — APPROVED** [review]

**Review by hipsterusername — APPROVED** [review]

**Comment by hipsterusername** [comment]
@sammcj - these are the tiny nudges and changes that keep the whole thing moving. Thanks for the help!!

**Comment by sammcj** [comment]
No worries at all, I don't want to come off as pushy / demanding so I'm glad it's taken that way 😄

**Pull request merged**